tests/test_data_model.py

```diff
@@ -858,3 +860,46 @@ def test_read_input_file(self):
         with self.subTest(msg="nonexistent file"):
             with self.assertRaises(FileNotFoundError):
                 data_model.read_input_file("nonexistent.json")
+
+    def test_mastodon_parsing(self):
+        mastodon_name = "a_tooter@example.exchange"
+        expected = HttpUrl("https://example.exchange/@a_tooter")
+        result = data_model.Connection.MastodonUname2Url(mastodon_name)
+        with self.subTest("Without '@' prefix"):
+            self.assertEqual(result, expected)
+
+        mastodon_name = "@a_tooter@example.exchange"
+        expected = HttpUrl("https://example.exchange/@a_tooter")
+        result = data_model.Connection.MastodonUname2Url(mastodon_name)
+        with self.subTest("With '@' prefix"):
+            self.assertEqual(result, expected)
+
+        mastodon_name = "@too@many@symbols"
+        with self.subTest("Too many '@' symbols"):
+            with self.assertRaises(ValueError):
+                data_model.Connection.MastodonUname2Url(mastodon_name)
+
+        mastodon_name = "@not_enough_at_symbols"
+        with self.subTest("Missing '@' separator"):
+            with self.assertRaises(ValueError):
+                data_model.Connection.MastodonUname2Url(mastodon_name)
+
+        mastodon_name = "user@bad_domain.example"
+        with self.subTest("Underscore in domain portion"):
+            with self.assertRaises(ValueError):
+                data_model.Connection.MastodonUname2Url(mastodon_name)
+
+        mastodon_name = "user@bad.numeric.tld.123"
+        with self.subTest("All digit TLD"):
+            with self.assertRaises(ValueError):
+                data_model.Connection.MastodonUname2Url(mastodon_name)
+
+        mastodon_name = "a_tooter@example.exchange."
+        expected = HttpUrl("https://example.exchange./@a_tooter")
+        result = data_model.Connection.MastodonUname2Url(mastodon_name)
+        with self.subTest("With FQDN root '.'"):
+            self.assertEqual(result, expected)
```

Comment:
This makes sense, and I will consider creating a new test for each social network. However, instead of using the `MastodonUname2Url` method directly, we can create a Connection object and check its `url` field (since we plan to move `MastodonUname2Url` to inside the `url` method.)

Reply:
Makes sense. Note that we can use Pydantic's HttpUrl class for the urls created for the other social networks.

Indeed, that offers use a simpler way of validating Mastodon domain names, as Pydantic does (mostly) check that the host part of a URL is syntactically valid.